test/test_fsdp.py

```diff
@@ -160,19 +184,26 @@ def run(mode: str, is_fp8: bool):
         model.load_state_dict(torch.load(sd_in_fname))
         optimizer = torch.optim.SGD(model.parameters(), lr=lr)
 
-        for _ in range(N_ITER):
+        def forward_backward():
```

Comment:
should this also respect the compile setting?

Reply:
I didn't enable compile for single-gpu because I wanted to use single-gpu without compile as the groundtruth to compare with the results (with or without compile) generated by multi-gpu to verify the numerics results. 
But I don't have a strong opinion about whether to respect the compile setting for single-gpu.